added new complex and action, updated changelog (#13041)
microsoftgraph/microsoft-graph-docs: api-reference/beta/resources/accesspackageassignmentrequest-requirements.md

---

> title: "accessPackageAssignmentRequestRequirements"

Reviewer: ```suggestion
title: "accessPackageAssignmentRequestRequirements resource type"
```

description: "The accessPackageAssignmentRequestRequirements type identifies the requirements necessary to request the specified access package."
localization_priority: Normal
author: "ajburnle"
ms.prod: "governance"
doc_type: "resourcePageType"
---

# accesspackageassignmentrequestrequirements

> **accessPackageAssignmentRequestRequirements** are requirements that a caller must fulfill in order to successfully create an accessPackageAssignmentRequest for the accessPackage specified as part of the URL. Requirements are determined by evaluating policies associated with the **accessPackage**. 

Reviewer: - "create an accessPackageAssignmentRequest" - change to a link to the API.
- Q: Is this API in any way related/dependent on what has been set in [accessPackageAssignmentPolicy](https://docs.microsoft.com/en-us/graph/api/resources/accesspackageassignmentpolicy?view=graph-rest-beta)?
    - I see references to "policies" but it's not clear which these policies are.
    - If there's any dependency, let's make that clear. If the properties below inherit from accessPackageAssignmentPolicy, let's also make that clear, even in their descriptions.
    - I see from the accesspackage-getapplicablepolicyrequirements.md, you say there's no request body, and there's also no create API to set them...to me, both states infer that the properties are inherited from somewhere.

```suggestion
In [Azure AD Entitlement Management](entitlementmanagement-root.md), access package assignment request requirements are requirements that a caller must fulfill in order to successfully create an accessPackageAssignmentRequest for an accessPackage. Requirements are determined by evaluating policies associated with the **accessPackage**. 
```

## Properties

This type has the following properties:

> [!NOTE]

> > All properties are not-required and effectively read-only because we have no API for a caller to create/modify one of these objects.

Reviewer: To remove.

| policyDisplayName | String | The display name of the policy that the user is trying to request access using. |
| policyDescription | String | The description of the policy that the user is trying to request access using.  |
| policyId | String | The ID of the policy that these requirements are associated with. This ID can be used when creating a new assignment request. |
| isApprovalRequired | Boolean | Whether or not a request must be approved by an approver. |
| isApprovalRequiredForExtension  | Boolean | Whether approval is required when a user tries to extend their access. |
| isRequestorJustificationRequired | Boolean | Whether a requestor must supply justification when submitting an assignment request. |
| isCustomAssignmentScheduleAllowed | Boolean | Whether the requestor is allowed to set a custom schedule. |
| schedule | requestSchedule | Schedule restrictions enforced, if any; null if none. |
| questions | accessPackageQuestion collection | Questions that are configured on the policy, required or optional; callers can determine if a question is required or optional based on the **isRequired** property on accessPackageQuestion. `null` if no questions are configured. |

> | existingAnswers | accessPackageQuestion collection | Answers that have already been provided. |

Reviewer: Previous comment on possible dependency/inheritance relationship applies here.

> 

Reviewer: ## Relationships

None

## JSON representation

The following is a JSON representation of the type.

```json
{ 
    "value": [ 
        { 
            "policyDisplayName": "string", 
            "policyDescription": "string",
            "policyId": "Edm.String",  
            "isApprovalRequired": false, 
            "isApprovalRequiredForExtension": false, 
            "isRequestorJustificationRequired": false,

>             "isCustomAssignmentScheduleAllowed": [],

Reviewer: ```suggestion
            "isCustomAssignmentScheduleAllowed": false,
```

---

> title: "accessPackageAssignmentRequestRequirements complex type"

Reviewer: File name looks off. Might be something like **accesspackageassignmentrequestrequirements.md**. Kindly check with the stub.

> description: "The accessPackageAssignmentRequestRequirements type identifies a collection requirements necessary to request the specified access package"

Reviewer: ```suggestion
description: "The accessPackageAssignmentRequestRequirements type identifies the requirements necessary to request the specified access package."
```

localization_priority: Normal
author: "ajburnle"
ms.prod: "governance"
doc_type: "resourcePageType"
---

# accessPackageAssignmentRequestRequirements complex type

> accessPackageAssignmentRequestRequirements are requirements that a caller must fulfill in order to successfully request an accessPackageAssignmentRequest for the accessPackage specified as part of the URL. Requirements are determined by evaluating policies associated with the accessPackage. 

Reviewer 2: ```suggestion
`accessPackageAssignmentRequestRequirements` are requirements that a caller must fulfill in order to successfully create an `accessPackageAssignmentRequest` for the `accessPackage` specified as part of the URL. Requirements are determined by evaluating policies associated with the `accessPackage`. 
```

> `accessPackageAssignmentRequestRequirements` are requirements that a caller must fulfill in order to successfully create an `accessPackageAssignmentRequest` for the `accessPackage` specified as part of the URL. Requirements are determined by evaluating policies associated with the `accessPackage`. 

Reviewer: Link accessPackageAssignmentRequest & accessPackage to the respective resources
```suggestion
**accessPackageAssignmentRequestRequirements** are requirements that a caller must fulfill in order to successfully create an accessPackageAssignmentRequest for the accessPackage specified as part of the URL. Requirements are determined by evaluating policies associated with the **accessPackage**. 
```

## Properties

This type has the following properties:

> [!NOTE]

> > All properties are not-required and effectively read-only because we have no API for a caller to create/modify one of these objects.

Reviewer 2: ```suggestion
```

I don't think this makes sense for public docs.

| Property                     | Type                      | Description |
| :--------------------------- | :------------------------ | :---------- |
| policyDisplayName | String | The display name of the policy that the user is trying to request access using. |
| policyDescription | String | The description of the policy that the user is trying to request access using.  |
| policyId | String | The ID of the policy that these requirements are associated with. This ID can be used when creating a new assignment request. |
| isApprovalRequired | Boolean | Whether or not a request must be approved by an approver. |
| isApprovalRequiredForExtension  | Boolean | Whether approval is required when a user tries to extend their access. |
| isRequestorJustificationRequired | Boolean | Whether a requestor must supply justification when submitting an assignment request. |
| isCustomAssignmentScheduleAllowed | Boolean | Whether the requestor is allowed to set a custom schedule. |

> | schedule | microsoft.graph.requestSchedule | Schedule restrictions enforced, if any; null if none. |
> | questions | Collection(microsoft.graph.accessPackageQuestion) | Questions that are configured on the policy, required or optional; callers can determine if a question is required or optional based on the isRequired property on accessPackageQuestion. Null if no questions are configured. |
> | existingAnswers | Collection(microsoft.graph.accessPackageQuestion) | Answers that have already been provided. |

Reviewer: Link requestSchedule , accessPackageQuestion to the respective resources.
```suggestion
| schedule | requestSchedule | Schedule restrictions enforced, if any; null if none. |
| questions | accessPackageQuestion collection | Questions that are configured on the policy, required or optional; callers can determine if a question is required or optional based on the **isRequired** property on accessPackageQuestion. `null` if no questions are configured. |
| existingAnswers | accessPackageQuestion collection | Answers that have already been provided. |
```

## JSON representation

The following is a JSON representation of the type.

> ```json

Reviewer 2: I think this blob has a bunch of unnecessary whitespace (maybe a side effect of copy/pasting with different line endings).

Author: removed white space

{ 

  

    "value": [ 

        { 

            "policyDisplayName": "string", 

            "policyDescription": "string",

            "policyId": "Edm.String",  

            "isApprovalRequired": false, 

            "isApprovalRequiredForExtension": false, 

            "isRequestorJustificationRequired": false,

            "isCustomAssignmentScheduleAllowed": [],

        } 

    ] 

> } 

Reviewer: + Remove whitespaces.
+ **isCustomAssignmentScheduleAllowed** property is a Boolean, so value is `false`.
+ 3 properties-**schedule**, **questions**, **existingAnswers**-missing here.
